### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by trevorflahardy**
Updated to have increased memory optimization and type checking for the end-user.

- Add a license to the top of the file.
- Add "dunder methods" to Account.
- Update documentation for these changes.

### CODE REVIEW

**Comment by trevorflahardy** [comment]
Note:

Did not mean to include both changes to `account.py` and `aes.py`, forgot to use a new branch. Please note there are two files that were affected in this PR. Will be fixed in any new PR's

File: fortnite_api/aes.py [hunk 1]
```diff
@@ -15,22 +42,23 @@ class AES:
...
     """
+    __slots__: Tuple[str, ...] = ('main_key', 'build', 'version', 'updated', 'dynamic_keys', 'raw_data')
 
     def __init__(self, data):
-        self.main_key = data.get('mainKey')
-        self.build = data.get('build')
-        self.version = re.search(r'\d\d.\d\d', self.build)[0] if self.build else None
+        self.main_key: str = data.get('mainKey')
+        self.build: str = data.get('build')
+        self.version = re.search(r'\d\d.\d\d', self.build)[0] if self.build else None # type: ignore
```
Comment (Luc1412) [inline fortnite_api/aes.py]: Why ignoring the type? It's a String.

Comment (Luc1412) [inline fortnite_api/aes.py]: Why ignoring the type? It's a String.

File: fortnite_api/aes.py [hunk 2]
```diff
@@ -40,8 +68,53 @@ def __str__(self):
...
+        if not isinstance(o, DynamicKey):
+            return False
+        
+        return self.pak_filename == o.pak_filename and self.pak_guid == o.pak_guid and self.key == o.key
+    
+    def __ne__(self, o: object) -> bool:
+        return not self.__eq__(o)
+    
+    def __str__(self) -> str:
+        return self.pak_filename
```
Comment (Luc1412) [inline fortnite_api/aes.py]: New line missing on the end of the file.

File: fortnite_api/account.py [hunk 3]
```diff
@@ -10,9 +55,29 @@ class Account:
     raw_data: :class:`dict`
         The raw data from request. Can be used for saving and re-creating the class.
     """
+    if TYPE_CHECKING:
+        external_auths: Dict # Adding when User lookup feature is enabled
```
Comment (Luc1412) [inline fortnite_api/account.py]: What's the reason for this change?

**Comment by trevorflahardy** [comment]
Fixed issues you mentioned before. 

`Account.external_auths` accidentally got marked as within type_hinting because I thought it got added to the class later. I thought it said "Added when User lookup feature is enabled" lol
